### PR TITLE
Custom replacement dictionaries for representation and bias tests

### PR DESCRIPTION
**Pull request by Prikshit7766**
## Checklist:

- [ ] I've added Google style docstrings to my code.
- [ ] I've used `pydantic` for typing when/where necessary.
- [ ] I have linted my code
- [ ] I have added tests to cover my changes.

#### Method: `pass_custom_data`

This method is used for the tasks of "bias" or "representation". It supports the following test_name:

### For bias:
- "Country-Economic-Bias"
- "Religion-Bias"
- "Ethnicity-Name-Bias"
- "Gender-Pronoun-Bias"

### For representation:
- "Country-Economic-Representation"
- "Religion-Representation"
- "Ethnicity-Representation"
- "Label-Representation" (only for NER)

### CODE REVIEW

File: langtest/transform/utils.py
Comment (JulesBelveze) [inline langtest/transform/utils.py:1]: For `RepresentationOperation` can you please:
- add types to the methods variables
- improve the docstrings by making sure that they all have a description, that the parameters are all described and that the methods are better documented

File: langtest/transform/custom_data.py [hunk 1]
```diff
@@ -13,23 +13,27 @@
 )
 
 
-def add_custom_data(data, name, append):
+def add_custom_data(data, name, append, task):
     """
```
Comment (JulesBelveze) [inline langtest/transform/custom_data.py:17]: Parameter data is never used within the function

File: langtest/transform/representation.py [hunk 2]
```diff
@@ -403,12 +399,16 @@ async def run(
...
+        )
 
         for sample in sample_list:
             if sample.test_type == "min_ethnicity_name_representation_proportion":
-                entity_representation_proportion = get_entity_representation_proportions(
-                    entity_representation
+                entity_representation_proportion = (
+                    RepresentationOperation.get_entity_representation_proportions(
+                        entity_representation
+                    )
```
Comment (JulesBelveze) [inline langtest/transform/representation.py:411]: I think there are extra parenthesis here and after

File: langtest/transform/utils.py
Comment (JulesBelveze) [reply langtest/transform/utils.py:1]: Also, make sure that you add the `@staticmethod` decorator to methods that do not have the `self` parameter

**Review by JulesBelveze — CHANGES_REQUESTED** [review]

File: langtest/transform/representation.py (hunk 2 above)
Comment (Prikshit7766) [reply langtest/transform/representation.py:411]: it is due to black formatting issues

**Comment by ArshaanNazir** [comment]
Can we add representation section as well in NB, showing both the default and custom replacement results then ? @Prikshit7766

**Comment by Prikshit7766** [comment]
we need to create separate  notebook for Custom replacement dictionaries for representation ?

**Comment by Prikshit7766** [comment]
@ArshaanNazir 
we need to create separate notebook for Custom replacement dictionaries for representation ?

**Comment by JulesBelveze** [comment]
@Prikshit7766 just add another section in the section

**Pull request merged**